lib/uplink/objects.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package uplink
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"storj.io/storj/pkg/storj"
+)
+
+// Object holds the information for a given object
+type Object struct {
+	Meta ObjectMeta
+}
+
+// ObjectMeta represents metadata about a specific Object
+type ObjectMeta struct {
+	Bucket   string
+	Path     storj.Path
+	IsPrefix bool
+
+	Metadata map[string]string
+
+	Created  time.Time
+	Modified time.Time
+	Expires  time.Time
+
+	Size     int64
+	Checksum []byte
+}
+
+// UploadOpts controls options about uploading a new Object, if authorized.
+type UploadOpts struct {
+	Metadata map[string]string
+	Expires  time.Time
+
+	Encryption *Encryption
+}
+
+// ListObjectsField numbers the fields of list objects
+type ListObjectsField int
+
+const (
+	// ListObjectsMetaNone opts
+	ListObjectsMetaNone ListObjectsField = 0
```

Comment:
you can just set this equal to iota and then not have any`=` for the following constants